src/morphology.rs

```diff
@@ -576,17 +577,39 @@ impl Mask {
     /// # }
     /// ```
     pub fn disk(radius: u8) -> Self {
-        let range = -(radius as i16)..=(radius as i16);
-        Self {
-            elements: range
+        let radius_squared = (radius as u32) * (radius as u32);
```

Comment:
Is there a reason why `clippy::cast_lossless` is an allowed lint? It's quite confusing when reading a lot of casts from `u8` to bigger types like `u16` and `u32` when such a cast should be lossless. It makes it harder to spot lossy casts when checking correctness.

Reply:
i know very little about clippy, i have no idea. i just run clippy and do what it tells me because i has always made sense so far. if i should change anything please tell me. in this specific function, i cast he `i16`s as `u32`s to prevent potential overflow from the squaring

Reply:
Apologies my question was more about this projects clippy settings and not your PR (it just happens to be the first PR I've noticed it on), if you're interested though you can read about the lint here: https://rust-lang.github.io/rust-clippy/master/index.html#/cast_loss. In this scenario I would recommend using `u32::from(radius)` or using `x.into()` methods over explicit casts. Ideally, clippy would enforce this for us however which was my wider point although that's not related to this particular PR.

Reply:
thanks, that makes a lot of sense, i've tried to use as many `from`s and `into()`s as possible. 

my next commit will take a little while because i've been working on he optimization of `erode` and `dilate` too.

Reply:
> Is there a reason why `clippy::cast_lossless` is an allowed lint? It's quite confusing when reading a lot of casts from `u8` to bigger types like `u16` and `u32` when such a cast should be lossless. It makes it harder to spot lossy casts when checking correctness.

I think the reasons are again the same as in the case of `unused imports`. Someone wrote code a long time ago that violates the rules, and it will take some time to rewrite it. Personally, I would remove all `allows` (in lib.rs), and where the rules still need to be violated, let it be localized to a function or module.

Reply:
If that is the case, I 100% agree with your approach. I can draft a PR to make that change but it would be quite a lot of code churn it might cause merge conflicts with some in-flight PRs. Perhaps after the next release if the PRs die down a bit we can give it a shot then.